cosmos/operators/watcher_kubernetes.py

```diff
@@ -0,0 +1,365 @@
+            airflow_version=AIRFLOW_VERSION,
+            dag_id=dag_id,
+            run_id=run_id,
+            producer_task_id=self.producer_task_id,
+            logger=logger,
+        )
+        if fetch_state is None:
+            return None
+
+        return fetch_state()
+
+    # slightly modified from the original DbtConsumerWatcherSensor to check try run and use K8s run
+    def execute(self, context: Context, **kwargs: Any) -> Any:
+        ti = context["ti"]
+        try_number = ti.try_number
+        if try_number > 1:
+            return self._fallback_to_k8s_run(try_number, context)
+        elif not self.poke(context):
+            self.defer(
+                trigger=WatcherTrigger(
+                    model_unique_id=self.model_unique_id,
+                    producer_task_id=self.producer_task_id,
+                    dag_id=self.dag_id,
+                    run_id=context["run_id"],
+                    map_index=context["task_instance"].map_index,
+                    use_event=False,
+                    poke_interval=self.poke_interval,
+                ),
+                timeout=self.execution_timeout,
+                method_name=self.execute_complete.__name__,
+            )
+        return None
+
+    def execute_complete(self, context: Context, event: dict[str, Any], **kwargs: Any) -> None:
+        status = event.get("status")
+        reason = event.get("reason")
+
+        if status != "failed":
+            return
+
+        if reason == "model_failed":
+            raise AirflowException(
+                f"dbt model '{self.model_unique_id}' failed. Review the producer task '{self.producer_task_id}' logs for details."
+            )
+
+        if reason == "producer_failed":
+            raise AirflowException(
+                f"Watcher producer task '{self.producer_task_id}' failed before reporting model results. Check its logs for the underlying error."
+            )
+
+    # def _use_event(self) -> bool:
+    #     if not self.invocation_mode:
+    #         self._discover_invocation_mode()
+    #     return self.invocation_mode == InvocationMode.DBT_RUNNER and EventMsg is not None
+
+    # minor change, removing event
+    def poke(self, context: Context) -> bool:
+        """
+        Checks the status of a dbt model run by pulling relevant XComs from the master task.
+        Handles retries and checks for successful completion of the model execution.
+        """
+        ti = context["ti"]
+        try_number = ti.try_number
+
+        logger.info(
+            "Try number #%s, poke attempt #%s: Pulling status from task_id '%s' for model '%s' and xcom '%s'",
+            try_number,
+            self.poke_retry_number,
+            self.producer_task_id,
+            self.model_unique_id,
+            f"{self.model_unique_id.replace('.', '__')}_status",
+        )
+
+        if try_number > 1:
+            return bool(self._fallback_to_local_run(try_number, context))
```

Comment:
```suggestion
            return bool(self._fallback_to_k8s_run(try_number, context))
```
I guess it should be this here?